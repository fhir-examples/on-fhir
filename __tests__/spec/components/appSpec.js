jest.mock('react-native-apple-healthkit', () => {
  return {
    initHealthKit: jest.fn((options, cb) => {
      cb(null, 1);
    })
  }
});

import React from 'react';

import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import HealthKit from 'react-native-apple-healthkit';

import App from '../../../app/app';

describe('AppSpec', () => {

  it('should initialize App (Router)', () => {

    // given
    const wrapper = shallow(<App />);

    // when
    wrapper.setState({ hkAvailable: true });

    // then
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });


  it('should not initialize App (Router), HealthKit unavailable', () => {

    // given
    const wrapper = shallow(<App />);

    // when
    wrapper.setState({ hkAvailable: false });

    // then
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });


  it('should initialize healthkit', () => {

    // given
    const wrapper = shallow(<App />);

    // when
    wrapper.instance().componentDidMount();

    // then
    expect(HealthKit.initHealthKit).toHaveBeenCalled();
  });


  it('should set hkAvailable to true', () => {

    // given
    const wrapper = shallow(<App />);

    // when
    wrapper.instance().componentDidMount();

    // then
    expect(wrapper.state().hkAvailable).toBe(true);
  });


  it('should set hkAvailable to false', () => {

    // given
    HealthKit.initHealthKit.mockImplementationOnce((opt, cb) => cb(true, null));

    // when
    const wrapper = shallow(<App />);
    wrapper.instance().componentDidMount();

    // then
    expect(wrapper.state().hkAvailable).toBe(false);
  });
});
