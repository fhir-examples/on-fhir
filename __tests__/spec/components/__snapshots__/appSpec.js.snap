// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSpec should initialize App (Router) 1`] = `
<Router
  barButtonIconStyle={
    Object {
      "tintColor": "#FFFFFF",
    }
  }
  barButtonTextStyle={
    Object {
      "color": "#FFFFFF",
    }
  }
  navigationBarStyle={
    Object {
      "backgroundColor": "#D81B60",
      "borderBottomColor": "#AD1457",
    }
  }
  titleStyle={
    Object {
      "color": "#FFFFFF",
    }
  }
>
  <_class
    unmountScenes={true}
  >
    <_class
      component={[Function]}
      initial={true}
      title="OnFhir"
    />
    <_class
      component={[Function]}
      title="OnFhir"
    />
  </_class>
</Router>
`;

exports[`AppSpec should not initialize App (Router), HealthKit unavailable 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    Error
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    HealthKit access denied!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    Give this app HealthKit permissions!
  </Text>
</View>
`;
